File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:
    stringLen = []

    def encode(self, strs: List[str]) -> str:
        outputStr = ""
        self.stringLen = []

        for s in strs: 
            self.stringLen.append(len(s))
            outputStr +=s
        return outputStr

    def decode(self, s: str) -> List[str]:
        output = []

        idx = 0 
        for slen in self.stringLen:
            output.append(s[idx:slen+idx])
            idx +=slen
        return output

File: Data Structures & Algorithms/reverse-bits/submission-0.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {

        uint32_t output = 0; 

        for (uint32_t i = 0; i < 32; i++)
        {
            output |= ((n & (1 << i)) >> i) << (31 - i);
        }

        return output; 
        
    }
};
